contracts/royalty-registry/src/helpers.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{
+    msg::{ExecuteMsg, QueryMsg, RoyaltyPaymentResponse},
+    state::RoyaltyEntry,
+    ContractError,
+};
+
+use cosmwasm_std::{ensure, to_binary, Addr, Deps, MessageInfo, QuerierWrapper, WasmMsg};
+use sg721_base::msg::{CollectionInfoResponse, QueryMsg as Sg721QueryMsg};
+use sg_std::Response;
+
+/// Ensures that the sender is the collection contract admin. If a collection contract admin does not exist,
+/// then the sender must be the collection contract creator.
```

Comment:
Hmm I'm sure it will have to change at some point to support other formats, but maybe that is not a problem. If creator cannot be retrieved it will throw an error and prevent execution.

Most sg721s support this query and they would all have to be upgraded to an incompatible format for this to break. In which case we upgrade royalty registry.

I'm not sure if it makes sense to write a complex integration test for this at this point.